vm/src/obj/objtuple.rs

```diff
@@ -292,27 +312,11 @@ pub fn init(context: &PyContext) {
 tuple(iterable) -> tuple initialized from iterable's items
 
 If the argument is a tuple, the return value is the same object.";
+    PyTuple::extend_class(context, tuple_type);
+
     extend_class!(context, tuple_type, {
-        "__add__" => context.new_rustfunc(PyTupleRef::add),
-        "__bool__" => context.new_rustfunc(PyTupleRef::bool),
-        "__eq__" => context.new_rustfunc(PyTupleRef::eq),
-        "__ne__" => context.new_rustfunc(PyTupleRef::ne),
-        "__contains__" => context.new_rustfunc(PyTupleRef::contains),
-        "__getitem__" => context.new_rustfunc(PyTupleRef::getitem),
-        "__hash__" => context.new_rustfunc(PyTupleRef::hash),
-        "__iter__" => context.new_rustfunc(PyTupleRef::iter),
-        "__len__" => context.new_rustfunc(PyTupleRef::len),
         (slot new) => tuple_new,
```

Comment:
`(slot new)` can be replaced by `#[pyslot(new)]`. See https://github.com/RustPython/RustPython/pull/1461/files#diff-8639ef334d679166adabaf89c18ffe6dR24